drawee/src/main/java/com/facebook/drawee/generic/GenericDraweeHierarchy.java

```diff
@@ -334,6 +336,13 @@ private DrawableParent getParentDrawableAtIndex(int index) {
   }
 
   /**
+   * Returns whether the given layer has a child drawable.
+   */
+  private boolean hasChildDrawableAtIndex(int index) {
+    return getParentDrawableAtIndex(index).getDrawable() != null;
```

Comment:
@s1rius Thanks for the change but it doesn't address what I noticed.

This calls through to `ArrayDrawable.getDrawableParentForIndex(int)` which will create the `ParentDrawable` against which you then extract the null child.

I think there are two better options:

1) add a method to `ArrayDrawable` called something like `hasDrawableAtIndex(int)` which just checks `mLayers[index]` for null and call that against `mFadeDrawable`; or
2) call `mFadeDrawable.getDrawable(int)` (as called for example in `setFailure(Throwable)`) which is equivalent to what you're doing here without have any side-effects

I realise that this approach existed previously in `hasPlaceholderImage()` but it would be great to improve on that while making this change. For bonus points, it looks like `setProgress(float)` could make use of this new method too.